### PR TITLE
Update the Cocoa backend to use a native fullscreen mode

### PR DESCRIPTION
**Pull request by perryprog**
The current fullscreen behavior for GZDoom on macOS is very odd, and does not behave like a native app (or game) would. This removes the older implementation of full-screening for macOS, and replaces it with the native Cocoa fullscreen toggle (`[NSWindow toggleFullscreen:]`). I also did a small amount of cleanup with the code surrounding the fullscreen implementation, but I feel like there's still a lot of housekeeping that could be done for the Cocoa backend.

As far as I can tell, everything still works as expected; I couldn't find any unexpected behavior when testing.

I'm also a little unhappy how this completely removes the old behavior—while I can't imagine why someone might want the old fullscreen implementation, I don't see why having an option could hurt (e.g. a menu option "Use native macOS fullscreen"). I decided not to put in anything additional yet, as I would like to see what a maintainer's comments on this would be.

### CODE REVIEW

**Comment by alexey-lysiuk** [comment]
There are two reasons for such fullscreen implementation.

Initially, we aimed to support 10.6, and it was extended to work on 10.4 PowerPC Macs.
Later, support for anything older than 10.9 was dropped, although there is no need to fix what's not broken.
The second reason is annoying spaces animation which require additional actions in order to disable it fully.

> I feel like there's still a lot of housekeeping that could be done for the Cocoa backend.

"Talk is cheap. Show me the code."

**Comment by perryprog** [comment]
>there is no need to fix what's not broken

I would argue that the fullscreen behavior is broken. It inadvertently makes it very difficult to switch between applications—something that I do quite often while playing—as well as causing a lot of other unexpected behavior with Spaces.

>annoying spaces animation

While I don’t find it annoying, I can see why it wouldn’t be preferred. However, if fullscreen is enabled within GZDoom’s settings, there is _no_ animation between choosing the IWAD and opening the main window in fullscreen mode. There will still be a (small) animation when toggling fullscreen mode.

>>I feel like there's still a lot of housekeeping that could be done for the Cocoa backend.

>"Talk is cheap. Show me the code."

Hah. I was specifically thinking of some weird formatting and a few warnings I was seeing from Clang-Tidy, but none of it was really anything besides visual differences. There’s nothing needed that I saw, and it wouldn’t be fixing anything that’s broken.

**Comment by perryprog** [comment]
Sounds good to me, your argument makes sense. I've gone ahead and done that, and it seems to work _mostly_ fine.

However, there is some odd behavior when changing the fullscreen preference while already in fullscreen mode. If you toggle the new option from non-native to native while in fullscreen mode, nothing will happen and you will be stuck in the non-native fullscreen mode until you toggle the new option again. If you toggle the new option from native to non-native while in fullscreen mode, you will gain a really funny/weird windowed mode while in fullscreen.

To clarify: toggling the new option while in fullscreen has some weird side-effects; I assume this should be prevented with a `CUSTOM_CVAR`. The callback needs to either prevent the option from being changed while in fullscreen, or it needs to disable fullscreen mode on the user's behalf. I'm not sure what way this should be done, but as far as I can tell, both seem doable with the callback you're supplied with when using `CUSTOM_CVAR`.

File: src/common/platform/posix/cocoa/i_video.mm [hunk 1]
```diff
@@ -102,6 +102,7 @@ - (void)enterFullscreenOnZoom
 EXTERN_CVAR(Bool, vk_debug)
 
 CVAR(Bool, mvk_debug, false, 0)
+CVAR(Bool, vid_macfullscreen, true, CVAR_ARCHIVE | CVAR_GLOBALCONFIG)
```
Comment (alexey-lysiuk) [inline src/common/platform/posix/cocoa/i_video.mm:105]: Please name it `vid_nativefullscreen` with the default value set to false.

Comment (perryprog) [reply src/common/platform/posix/cocoa/i_video.mm:105]: Ah sorry, I thought you had said to have it enabled by default in your first comment.

**Comment by perryprog** [comment]
I'm not sure I follow, what extra changes are you referring to? I haven't added anything since your first review. If you mean the update from the old mask enum to the new one, I can change that back. I left it in there since it was relevant to what this PR was modifying, and it didn't hurt.

I didn't realize you had originally meant this did not need a menu entry, so I'll change it to just be a CVAR.

I have no idea how those two bits got formatted with spaces—I double checked that there weren't any before committing, but I guess it got reformatted anyway; sorry about that. I also see that there was another accidental format change with my last commit, which I will also change back.

Since this is going to now be three/four commits, let me know if you want me to squash it to one commit in case you prefer to do a fast-forward instead of a merge.

**Comment by alexey-lysiuk** [comment]
> I'm not sure I follow, what extra changes are you referring to? I haven't added anything since your first review. If you mean the update from the old mask enum to the new one, I can change that back. I left it in there since it was relevant to what this PR was modifying, and it didn't hurt.

I’m referring to changed signature of `CreateWindow()` function, to modified window style constants, and to addition of `[window exitAppOnClose];` line. Can it be done without all these changes?

**Comment by perryprog** [comment]
Ah, ok. Yes, this change can be done without any of those changes. All of those were more stylistic choices, except for the window style constants which were deprecated. Additionally, the `[window exitAppOnClose]` line was moved from `SetWindowedMode()` as it only needed to be called during the initialization of the window. I accidentally added it back when I reverted by removal of `SetWindowedMode()`.

Since I assume you're implying that you want those changes reverted, I'm going to go ahead and undo them except for `[window exitAppOnClose]` as that change is closer to a bugfix than a style choice.

I'm sorry for any confusion I might have caused from that.

**Pull request merged**